src/Content/Conversation/Entity/Timeline.php

```diff
@@ -22,11 +22,15 @@
 namespace Friendica\Content\Conversation\Entity;
 
 /**
- * @property-read string $code        Channel code
- * @property-read string $label       Channel label
- * @property-read string $description Channel description
- * @property-read string $accessKey   Access key
- * @property-read string $path        Path
+ * @property-read string $code           Channel code
+ * @property-read string $label          Channel label
+ * @property-read string $description    Channel description
+ * @property-read string $accessKey      Access key
+ * @property-read string $path           Path
+ * @property-read int    $uid            User of the channel
+ * @property-read string $includeTags    The tags to include in the channel
+ * @property-read string $excludeTags    The tags to exclude in the channel
+ * @property-read string $fullTextSearch full text search pattern
  */
 final class Timeline extends \Friendica\BaseEntity
```

Comment:
Shouldn't it be renamed to `Channel` now that it is linked to the `channel` database table? Or separated in a different class?

Reply:
The user defined channels are a part of the channels that are part of all the timelines. I see that table only as a part of the whole functionality.

Reply:
Sure, but in this case would you be able to add a `Entity\Timeline` parameter to `Conversation/Timeline->getRawChannelItems()` in order to build the condition?

On the other hand, you would have to retrieve said `Entity\Timeline` object in one place whether it is a built-in timeline, channel or user-defined channel.

This way I believe it's easier to figure it out if all those concepts can be boiled down to a single `Entity\Timeline` object that can be used to figure the condition to retrieve items.

Reply:
>  the entity class could be filled via fixed values (the currently hard coded channels), via the user (the functionality in this PR) and maybe in the future via the admin or by subscribing to some external service.

And this is fine, since the Entity concept isn't coupled with a database only through a Repository and a `ICanCreateFromTableRow` factory.

What I want to get at is that the resulting Channel/Timeline objects should be used the same way. So ideally the `Conversation\Timeline->getRawItems()` should take a Channel/Timeline object as a parameter and determine the conditions to fetch the items based on this object property alone, not based on `selectedTab` for example.

This may force you to add more fields to the corresponding entity, and it's fine.

Reply:
In the future I would like to make the channels available via the API as well. So the question will be, how to achieve this. But I would like to go an iterative way here. When this PR isn't wrong from an architectural point of view, I would prefer having it merged now, so that I can focus on stuff like bugs and UX for the upcoming release.